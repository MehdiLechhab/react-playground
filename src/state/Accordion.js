import React from "react";

class Accordion extends React.Component {
	static defaultProps = { 
		sections: [] 
	};
	state = {
    	currentSectionIndex: null
  	};

  	handleButtonClick = (index) => {
    console.log('button clicked!', { index })
    this.setState({ currentSectionIndex: index })

  }


  	renderButton(section, idx, currentIndex) {
  		return (
  			<li key={idx}>
  				<button onClick={() => this.handleButtonClick(idx)}>{section.title}</button>
  				{(currentIndex === idx) && <p>{section.content}</p>}
  			</li>
  		)
  	}
 
  render() {
  	const {currentSectionIndex} = this.state
  	const {sections} = this.props
    return (
      <ul>
        {sections.map((section, idx) =>
        	this.renderButton(section, idx, currentSectionIndex)
        	)}
      </ul>
    )
  }
}


export default Accordion;