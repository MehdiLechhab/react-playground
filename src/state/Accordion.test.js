import React from 'react';
import ReactDOM from 'react-dom';
import Accordion from './Accordion';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'


describe(`Sections Component`, () => {
  // array of objects, each with a name and content
  const sections = [
  {
    title: 'Section 1',
    content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
  },
  {
    title: 'Section 2',
    content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
  },
  {
    title: 'Section 3',
    content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
  },
]

  it('renders without errors', () => {
    const li = document.createElement('li');
    ReactDOM.render(<Accordion />, li);
    ReactDOM.unmountComponentAtNode(li);
 	})

  it('renders empty given no section', () => {
    const wrapper = shallow(<Accordion />)
    expect(toJson(wrapper)).toMatchSnapshot()
 	})

  it('renders no section by default', () => {
    const wrapper = shallow(<Accordion sections={sections} />)
    expect(toJson(wrapper)).toMatchSnapshot()
 	})

  it('closes the first sections and opens the clicked section', () => {
	const wrapper = shallow(<Accordion sections={sections} />)
	wrapper.find('button').at(1).simulate('click')
  	expect(toJson(wrapper)).toMatchSnapshot()
	})

  
})