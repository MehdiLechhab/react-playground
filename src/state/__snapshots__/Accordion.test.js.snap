// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sections Component closes the first sections and opens the clicked section 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
    <p>
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </p>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Sections Component renders empty given no section 1`] = `<ul />`;

exports[`Sections Component renders no section by default 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;
